project/MimaExcludes.scala

```diff
@@ -465,6 +465,9 @@ object MimaExcludes {
 
   // Exclude rules for 2.2.x
   lazy val v22excludes = v21excludes ++ Seq(
+    // [SPARK-19591][ML] Add sample weights to decision trees
+    ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.mllib.tree.configuration.Strategy.this"),
```

Comment:
This is under the 2.2 rules but should be under 3.0. We could also keep this constructor, but I don't think it's worth it.

Reply:
good catch, fixed!